### PR TITLE
Refactor session class

### PR DESCRIPTION
**Pull request by BakerWJ**

### CODE REVIEW

**Comment by BakerWJ** [comment]
Changed name from Session to Section, removed builder for Section, and redid CourseCreator with updated implementation

**Review by roryhofland — COMMENTED** [review]
Just tried `CommandLineInterface` on some classes, looks like its working well. I suppose all that needs to be added now are new refactored tests, but thats pretty much what the mock API pr is for.

**Comment by BakerWJ** [comment]
> Just tried `CommandLineInterface` on some classes, looks like its working well. I suppose all that needs to be added now are new refactored tests, but thats pretty much what the mock API pr is for.

I updated the tests for CourseCreator so they're working - the issue is the tests for scheduler and the other stuff. I also didn't really make any changes to CommandLineInterface so there are some issues with it right now if you try to use a course that is only in the winter.

File: src/main/java/CourseCreator.java [hunk 1]
```diff
@@ -3,95 +3,86 @@
...
 
     /**
-     * Generates a course given a courseId.
+     * Generates a course given a courseID and the session of the course
      *
-     * @param courseId the id of the course to be created.
+     * @param courseId the identifier for hte course
+     * @param session the semester that the course takes place ie S, F, Y
      * @return a Course object
+     * @throws IOException i dont know what exception this throws lol
```
Comment (EclipseIsDead) [inline src/main/java/CourseCreator.java:30]: If the API doesn't work it should throw IO exception, so you use it any time you create a API Woker

**Review by EclipseIsDead — APPROVED** [review]

**Review by kenneth-miura — APPROVED** [review]
lgtm

**Pull request merged**